monai/transforms/croppad/array.py

```diff
@@ -644,15 +697,19 @@ def set_random_state(
     def randomize(self, data: Optional[Any] = None) -> None:
         pass
 
-    def __call__(self, img: NdarrayOrTensor) -> List[NdarrayOrTensor]:
+    def __call__(self, img: torch.Tensor) -> List[torch.Tensor]:
         """
         Apply the transform to `img`, assuming `img` is channel-first and
         cropping doesn't change the channel dim.
         """
-        return [self.cropper(img) for _ in range(self.num_samples)]
+        ret = [self.cropper(img) for _ in range(self.num_samples)]
```

Comment:
I think this will require deepcopy of meta dict?

Reply:
Hi @wyli ,

Actually, I asked you about this problem offline, you told me `MetaTensor` will automatically deepcopy when cropping if not a batch data?
Let me add the deepcopy logic if necessary.

Thanks.

Reply:
By default it'll be automatic but in this pr you have made most changes in-plcae as you commented earlier https://github.com/Project-MONAI/MONAI/pull/4548#issuecomment-1161268944

Reply:
Hi @wyli ,

I think the `slices` operation on a `MetaTensor` in the crop transforms will automatically deepcopy the content (including meta), just as we discussed offline before.
I added more unit tests to cover it.
It also can be verified by below program:
```py
from monai.utils import convert_to_tensor

a = convert_to_tensor([1, 2, 3], track_meta=True, wrap_sequence=True)
b = a[1:]
b.meta["patch_index"] = 1
b.affine[0, 1] = 2
```
```
>>> print(b)
tensor([2, 3])
MetaData
        affine: tensor([[1., 2., 0., 0.],
        [0., 1., 0., 0.],
        [0., 0., 1., 0.],
        [0., 0., 0., 1.]], dtype=torch.float64)
        patch_index: 1

Applied operations

Is batch?: False
>>> 
>>> print(a)
tensor([1, 2, 3])
MetaData
        affine: tensor([[1., 0., 0., 0.],
        [0., 1., 0., 0.],
        [0., 0., 1., 0.],
        [0., 0., 0., 1.]], dtype=torch.float64)

Applied operations

Is batch?: False
>>>
```
What do you think?

Thanks.

Reply:
I see, it looks good here.